pkg/k8shandler/cluster.go

```diff
@@ -274,3 +306,89 @@ func (cState *ClusterState) amendPods(dpl *v1alpha1.Elasticsearch) error {
 	}
 	return nil
 }
+
+func updateClusterSettings(dpl *v1alpha1.Elasticsearch) error {
+
+	event, err := getClusterEvent(dpl)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Event is: %s\n", event)
+
+	switch event {
+	case v1alpha1.ScaledDown:
+		if err := execClusterSettingsUpdate(dpl); err != nil {
+			return err
+		}
+		utils.UpdateClusterSettingsUpdateNeededCondition(&dpl.Status, v1alpha1.ConditionFalse)
+		utils.UpdateScalingDownCondition(&dpl.Status, v1alpha1.ConditionFalse)
+		break
+	case v1alpha1.ScaledUp:
+		if err := execClusterSettingsUpdate(dpl); err != nil {
+			return err
+		}
+		utils.UpdateClusterSettingsUpdateNeededCondition(&dpl.Status, v1alpha1.ConditionFalse)
+		utils.UpdateScalingUpCondition(&dpl.Status, v1alpha1.ConditionFalse)
+		break
+	case v1alpha1.NoEvent:
+		return nil
+	}
+
+	if err = sdk.Update(dpl); err != nil {
+		return fmt.Errorf("Unable to update Elasticsearch cluster status: %v", err)
+	}
+
+	return nil
+}
+
+// TODO: this method name should be more descriptive
+func getClusterEvent(dpl *v1alpha1.Elasticsearch) (v1alpha1.ClusterEvent, error) {
+	_, settingsUpdateCondition := utils.GetESNodeCondition(&dpl.Status, v1alpha1.ClusterSettingsUpdateNeeded)
+	_, scaleDownCondition := utils.GetESNodeCondition(&dpl.Status, v1alpha1.ScalingDown)
+	_, scaleUpCondition := utils.GetESNodeCondition(&dpl.Status, v1alpha1.ScalingUp)
+
+	if settingsUpdateCondition != nil && settingsUpdateCondition.Status == v1alpha1.ConditionTrue {
+		if scaleDownCondition != nil && scaleDownCondition.Status == v1alpha1.ConditionTrue {
+			// TODO: here we can wait till no pods are Terminating
+			return v1alpha1.ScaledDown, nil
+		}
+		if scaleUpCondition != nil && scaleUpCondition.Status == v1alpha1.ConditionTrue {
+			pods, err := listPods(dpl.Name, dpl.Namespace)
```

Comment:
do you somehow need to wait for all expected pods?  e.g. what if `listPods` returns only 3 pods but because of the scaleup now 5 are expected?

Reply:
@richm I changed it so that the number of currently running nodes is read from `es-logging:9200/_cluster/health["number_of_nodes"]`

What do you think?

Reply:
> @richm I changed it so that the number of currently running nodes is read from es-logging:9200/_cluster/health["number_of_nodes"]

That could still return less than the expected number of nodes

Reply:
Yes, we wait for all nodes to join the cluster before we mark the scaleUp operation as complete.
Similarly for scaleDown, except those nodes leave the cluster.

https://github.com/openshift/elasticsearch-operator/pull/33/files#diff-b0f11234653f638f62ef2fc4f695a69dR380